infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -235,6 +226,23 @@ public synchronized void refresh(@Nonnull TopologyDescriptor newTopology) {
         if (replicationContext.getIsLeader().get()) {
             removeStaleSessionOnLeadershipAcquire();
         }
+
+        // check if any new SINKs have been added to the topology.
+        Set<String> newRemoteSinksAdded = Sets.difference(newRemoteSinks, currentRemoteSinks);
+        if (!newRemoteSinksAdded.isEmpty()) {
+            configManager.getRegisteredSubscribers().forEach(this::createSessions);
```

Comment:
we should account for remoteSink -> replicationModel mapping for creating these sessions instead of creating them for all models.